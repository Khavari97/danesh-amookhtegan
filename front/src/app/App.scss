@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


