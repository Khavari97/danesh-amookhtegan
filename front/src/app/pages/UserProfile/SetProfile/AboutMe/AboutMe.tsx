import React, {Component} from 'react';
import Style from './../SetProfile.module.scss'
import {Grid, ListItem} from "@material-ui/core";
import {connect} from 'react-redux'
import {GetProfile} from "../../../../redux/User/Profile/Actions";
import {rootState} from "../../../../redux/RootReducer";
import Loading from "./../../../../components/customs/Loading/Loading";

interface IState {
    fields?: object | any;
}

interface IProps {
    GetProfile: typeof GetProfile
    description: string
    data: any
    loading: boolean
}

const mapStateToProps = (state: rootState) => {
    return {
        province: state.userProfile.provinces,
        data: state.userProfile.data,
        description: state.userProfile.description,
        loading: state.userProfile.loading
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        GetProfile: () => dispatch(GetProfile())

    }
}

class AboutMe extends Component<IProps, IState> {
    render() {
        return (
            <Grid container className="animate__animated animate__fadeIn">
                {(this.props.loading) ?
                    <Loading/>
                    :
                    <Grid item xs={12} sm={12}>
                        <Grid className={Style.gridItems} item xs={12} sm={12}>
                            <ListItem><p data-testid='desc' className={Style.title}>{this.props.description}</p></ListItem>
                        </Grid>
                    </Grid>
                }
            </Grid>

        );
    }

}

export default connect(mapStateToProps, mapDispatchToProps)(AboutMe)