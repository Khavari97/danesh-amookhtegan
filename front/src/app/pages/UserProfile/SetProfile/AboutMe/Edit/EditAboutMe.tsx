import React, {Component} from 'react';
import {Grid, ListItem, TextareaAutosize} from "@material-ui/core";
import Style from '../../SetProfile.module.scss'
import {Formik} from "formik";
import {connect} from 'react-redux'
import {GetProfile, ProvinceApi} from "../../../../../redux/User/Profile/Actions";
import {rootState} from "../../../../../redux/RootReducer";
import {apiClient} from './../../../../../api/_api'

interface IProps {
    forwardedRef: any
    ProvinceApi: any
    province: any
    GetProfile: typeof GetProfile
    description: string
    data: any
}

const mapStateToProps = (state: rootState) => {
    return {
        province: state.userProfile.provinces,
        data: state.userProfile.data,
        description: state.userProfile.description
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        ProvinceApi: () => dispatch(ProvinceApi()),
        GetProfile: () => dispatch(GetProfile())

    }
}

class EditAboutMe extends Component<IProps, any> {

    postData = (values: any) => {
        let data = {
            firstName: this.props.data.firstName,
            lastName: this.props.data.lastName,
            email: this.props.data.email,
            username: this.props.data.username,
            city: this.props.data.cityObject,
            university: this.props.data.universityObject,
            phoneNumber: this.props.data.phoneNumber,
            description: values.description
        }
        apiClient.patch('user/profile/', {
            data: data,
        })
            .then(() => {
                this.props.GetProfile()
            })
    }

    render() {
        return (
            <Formik
                innerRef={this.props.forwardedRef}
                initialValues={{description: this.props.description}}
                onSubmit={(values, actions) => {
                    this.postData(values)
                    // actions.setSubmitting(false);
                }}
            >
                {props => (
                    <form onSubmit={props.handleSubmit}>
                        <Grid container className="animate__animated animate__fadeIn">
                            <Grid item xs={12} sm={12}>
                                <Grid className={Style.gridItems} item xs={12} sm={12}>
                                    <ListItem>
                                        <TextareaAutosize name='description' value={props.values.description}
                                                          onChange={props.handleChange}
                                                          aria-label="minimum height"
                                                          data-testid="description"
                                                          style={{width: '100%', fontSize: '20px'}} rowsMin={10}/>
                                    </ListItem>
                                </Grid>
                            </Grid>
                        </Grid>
                    </form>
                )}
            </Formik>

        );
    }

}

export default connect(mapStateToProps, mapDispatchToProps)(EditAboutMe)