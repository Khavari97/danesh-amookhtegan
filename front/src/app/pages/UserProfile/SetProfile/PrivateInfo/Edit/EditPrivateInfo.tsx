import React, {Component} from 'react';
import Style from './../../SetProfile.module.scss'
import {Button} from "../../../../../components/material-ui"
import {Formik, FormikProps, FormikValues} from "formik";
import CustomInputs from "../../../../../components/customs/input/input";
import {
    Dialog,
    DialogActions,
    DialogContent,
    DialogContentText,
    DialogTitle,
    Grid,
    IconButton,
    ListItem
} from '@material-ui/core';
import images from './../../../../../assets/images/userProfile'
import Select from './../../../../../components/customs/select/select';
import {connect} from 'react-redux'
import {GetProfile, ProvinceApi} from "../../../../../redux/User/Profile/Actions";
import {rootState} from "../../../../../redux/RootReducer";
import {apiClient} from './../../../../../api/_api'
import * as yup from "yup";
import Snackbar from "./../../../../../components/material-ui/Snackbar/SnackbarUtils";

interface IState {
    fields?: object | any;
    emailModal: boolean;
    initial: object | any

}

interface IProps {
    forwardedRef: any
    provinces: any
    cities: any
    GetProfile: typeof GetProfile
    universities: any
    data: any
    description: string
    postData?:any
}

const mapStateToProps = (state: rootState) => {
    return {
        provinces: state.userProfile.provinces,
        universities: state.userProfile.universities,
        cities: state.userProfile.cities,
        data: state.userProfile.data,
        description: state.userProfile.description
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        ProvinceApi: () => dispatch(ProvinceApi()),
        GetProfile: () => dispatch(GetProfile())
    }
}


class PrivateInfo extends Component<IProps, IState> {
    private formRef: any;

    constructor(props: IProps) {
        super(props);
        this.formRef = React.createRef()
        this.state = {
            fields: [
                [{
                    id: 'signInputs',
                    type: "text",
                    name: "firstName",
                    label: "نام",
                    placeholder: "",
                },
                    {
                        id: 'signInputs',
                        type: "text",
                        name: "lastName",
                        label: "نام خانوادگی",
                        placeholder: "",
                    },
                ],
                {
                    id: 'signInputs',
                    type: "text",
                    name: "gender",
                    label: "جنسیت",
                    items: [{name: "مرد", id: "MALE"}, {name: "زن", id: "FEMALE"}],
                    placeholder: "",
                },
                {
                    id: 'signInputs',
                    type: "text",
                    name: "username",
                    label: "نام کاربری",
                    placeholder: "",
                },
                {
                    id: 'signInputs',
                    type: "email",
                    name: "email",
                    label: "ایمیل",
                    placeholder: "",
                    value: ''
                },
                [
                    {
                        id: 'signInputs',
                        type: "text",
                        name: "province",
                        label: "استان",
                        placeholder: "",
                        items: this.props.provinces,
                    },
                    {
                        id: 'signInputs',
                        type: "text",
                        name: "city",
                        label: "شهر",
                        placeholder: "",
                        items: this.props.cities,
                    },
                ],
                {
                    id: 'signInputs',
                    type: "text",
                    name: "university",
                    label: "دانشگاه",
                    placeholder: "",
                    items: this.props.universities,
                },
                {
                    id: 'signInputs',
                    type: "text",
                    name: "phoneNumber",
                    label: "شماره تلفن",
                    placeholder: "",
                    value: '+98'
                },
            ],
            emailModal: false,
            initial: {
                firstName: this.props.data?.firstName,
                lastName: this.props.data?.lastName,
                email: this.props.data?.email,
                username: this.props?.data.username,
                province: this.props?.data.provinceObject,
                city: this.props.data?.cityObject,
                university: this.props.data?.universityObject,
                phoneNumber: (this.props.data?.phoneNumber == null || this.props.data.phoneNumber == '')
                    ? '+98' : this.props.data?.phoneNumber,
                description: this.props?.description,
                gender: this.props.data?.gender
            }
        }
    }

    componentDidUpdate() {
        if (this.state.fields[4][1].items != undefined) {
            if (!this.arraysAreEqual(this.props.cities, this.state.fields[4][1].items)) // Check if it's a new user, you can also use some unique property, like the ID  (this.props.user.id !== prevProps.user.id)
            {
                let fields = [...this.state.fields]
                fields[4][1].items = this.props.cities
                this.setState({fields: fields})
            }
        }
    }

    arraysAreEqual = (ary1: any, ary2: any) => {
        return (ary1.join('') == ary2.join(''));
    }
    handleModalOpen = () => {
        this.setState({emailModal: !this.state.emailModal})
    }

    emailDialog = () => {
        return (
            <div>
                <IconButton aria-label="delete" onClick={this.handleModalOpen}>
                    <img style={{width: '20px'}} alt="" src={images.alert}/>
                </IconButton>
                <Dialog
                    open={this.state.emailModal}
                    onClose={this.handleModalOpen}
                    aria-labelledby="alert-dialog-title"
                    aria-describedby="alert-dialog-description"
                >
                    <DialogTitle id="alert-dialog-title">{" توضیحات"}</DialogTitle>
                    <DialogContent>
                        <DialogContentText id="alert-dialog-description">
                            {" ایمیل شما به دلیل تایید و ذخیره در سایت غیر قابل تغییر است"}
                        </DialogContentText>
                    </DialogContent>
                    <DialogActions>
                        <Button variant="contained" onClick={this.handleModalOpen} color="primary">
                            فهمیدم
                        </Button>
                    </DialogActions>
                </Dialog>
            </div>
        )
    }


    selectItems = (element: any, props: FormikProps<any>) => {
        return (
            <Select object={{
                ...element,
                data_test:element.name,
                value: props.values[element.name],
                setFieldValue: props.setFieldValue,
                onChange: props.handleChange
            }}></Select>
        )
    }


    inputs = (element: any, index: string, props: FormikProps<any>) => {
        if (element instanceof Array) {
            return element.map((item: any, index2: number) => {
                if (item.items !== undefined) {
                    return (
                        <Grid item className={Style.gridInputs} xs={12} md={5} key={index + index2}>
                            {this.selectItems(item, props)}
                        </Grid>
                    )

                } else {
                    return (
                        <Grid item className={Style.gridInputs} xs={12} md={5} key={index + index2}>
                            <CustomInputs items={{
                                ...item,
                                data_test:item.name,
                                error: Boolean(props.errors[item.name]),
                                helperText: (props.errors[item.name]),
                                value: props.values[item.name],
                                onChange: props.handleChange
                            }}/>
                        </Grid>
                    )
                }


            })
        } else {
            if (element.name === 'email') {
                return (
                    <React.Fragment key={index}>
                        <div key={index} data-testid={element.name}>
                            <ListItem><p className={Style.title}>{element.label}</p></ListItem>
                            <ListItem><p className={Style.description}>{this.props.data.email}</p></ListItem>
                        </div>
                        {this.emailDialog()}
                    </React.Fragment>
                )
            }
            if (element.items !== undefined) {
                return (
                    <Grid item className={Style.gridInputs} xs={12} md={5} key={index}>
                        {this.selectItems(element, props)}
                    </Grid>
                )

            } else {
                return (
                    <Grid xs={12} md={5} className={Style.gridInputs} item key={index}>
                        <CustomInputs items={{
                            ...element,
                            data_test:element.name,
                            error: Boolean(props.errors[element.name]),
                            helperText: (props.errors[element.name]),
                            value: props.values[element.name],
                            onChange: props.handleChange
                        }}/>
                    </Grid>
                )
            }


        }
    }

    items = (props: FormikProps<any>) => {
        return this.state.fields.map((element: any, index: number) => {
            return (
                <Grid key={index} style={{display: 'flex'}} container className={Style.gridItems} item xs={12} sm={6}>
                    {this.inputs(element, 'input' + index, props)}
                </Grid>

            )
        })
    }

    postData = (values: FormikValues) => {
        let data = {
            firstName: values.firstName,
            lastName: values.lastName,
            username: values.username,
            email: values.email,
            university: (values.university) ? values.university : null,
            city: (values.city) ? values.city : null,
            phoneNumber: (values.phoneNumber == '') ? null
                : (values.phoneNumber.startsWith('+98')) ? ((values.phoneNumber.substring(3) == '') ? null
                    : values.phoneNumber) :
                    '+98' + ((values.phoneNumber.startsWith('0')) ? values.phoneNumber :
                    values.phoneNumber.substring(0)),
            description: this.props.description,
            gender: (values.gender) ? values.gender : null,
        }
        apiClient.patch('user/profile/', {data: data})
            .then(() => {
                this.props.GetProfile()
            })
            .catch((err) => {
                console.log(err)
                if (err.status == 400) {
                    for (const [key, value] of Object.entries(err.data.detail)) {
                        Snackbar.error('' + value)
                    }
                }
            })
    }


    render() {
        return (
            <Formik innerRef={this.props.forwardedRef}
                    enableReinitialize
                    initialValues={this.state.initial}
                    validationSchema={yup.object({
                        firstName: yup
                            .string()
                            .required("پر کردن این فیلد اجباری است")
                            .max(30, "حداکثر 30 حرف میتوانید وارد کنید")
                            .min(1, "حداقل 1 حرف میتوانید وارد کنید")
                            .matches(/^[\u0600-\u06FF\s]+$/, "تنها استفاده از حروف فارسی مجاز است"),
                        lastName: yup
                            .string()
                            .required("پر کردن این فیلد اجباری است")
                            .max(30, "حداکثر 30 حرف میتوانید وارد کنید")
                            .min(1, "حداقل 1 حرف میتوانید وارد کنید")
                            .matches(/^[\u0600-\u06FF\s]+$/, "تنها استفاده از حروف فارسی مجاز است"),
                        username: yup
                            .string()
                            .required("پر کردن این فیلد اجباری است")
                            .max(30, "حداکثر 30 حرف میتوانید وارد کنید")
                            .min(1, "حداقل 1 حرف میتوانید وارد کنید")
                            .matches(/^[a-zA-Z0-9-_]+$/, "تنها استفاده از حروف انگلیسی مجاز است"),
                        phoneNumber: yup
                            .string()
                            .nullable(true)
                            .matches(/^[+]*[(]{0,1}[0-9]{1,3}[)]{0,1}[-\s\./0-9]*$/g, "شماره ی معتبر وارد کنید"),
                    })}
                    onSubmit={(values, actions) => {
                            this.props.postData(values)
                            this.postData(values)
                            actions.setSubmitting(false);
                    }}
            >
                {props => (
                    <form className={Style.formContainer} onSubmit={props.handleSubmit}>
                        <Grid className="animate__animated animate__fadeIn" container>
                            {this.items(props)}
                        </Grid>
                    </form>
                )}
            </Formik>

        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(PrivateInfo)
