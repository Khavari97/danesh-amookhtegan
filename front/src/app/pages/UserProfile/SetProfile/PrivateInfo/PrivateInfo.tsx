import React, {Component} from 'react';
import Grid from '@material-ui/core/Grid';
import Style from './../SetProfile.module.scss'
import {ListItem} from "@material-ui/core";
import {connect} from 'react-redux'
import {rootState} from "../../../../redux/RootReducer";
import {GetProfile} from "../../../../redux/User/Profile/Actions";
import Loading from "./../../../../components/customs/Loading/Loading";

interface IState {
    fields?: object | any;
}

interface IProps {
    description: string
    data: any
    GetProfile: typeof GetProfile
    loading: boolean
}


const mapStateToProps = (state: rootState) => {
    return {
        province: state.userProfile.provinces,
        data: state.userProfile.data,
        description: state.userProfile.description,
        loading: state.userProfile.loading
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        GetProfile: () => dispatch(GetProfile())
    }
}

class PrivateInfo extends Component<IProps, IState> {

    constructor(props: IProps) {
        super(props);

        this.state = {
            fields: [
                [{
                    title: 'نام',
                    name: 'firstName',

                },
                    {
                        title: 'نام خانوادگی',
                        name: 'lastName',

                    }
                ],
                {
                    title: 'جنسیت',
                    name: 'genderDisplay',

                },
                {
                    title: 'نام کاربری',
                    name: 'username',

                },
                {
                    title: 'ایمیل',
                    name: 'email',

                },
                [{
                    title: 'استان',
                    name: 'province',
                    value: 'تهران'
                },
                    {
                        title: 'شهر',
                        name: 'city',

                    }],

                {
                    title: 'دانشگاه',
                    name: 'university',

                },
                {
                    title: 'شماره تلفن',
                    name: 'phoneNumber',

                }
            ]
        }
    }

    componentDidMount() {
        //this.props.GetProfile()
    }

    inputs = (element: any, index: number) => {
        if (element instanceof Array) {
            return element.map((item: any, index2: number) => {
                return (
                    <div key={index2}>
                        <ListItem><p className={Style.title}>{item.title}</p></ListItem>
                        <ListItem><p data-testid={item.name} className={Style.description}>{this.props?.data[item.name]}</p></ListItem>
                    </div>
                )
            })
        } else {
            return (
                <div key={index}><ListItem><p className={Style.title}>{element.title}</p></ListItem>
                    <ListItem><p data-testid={element.name}  className={Style.description}>{this.props?.data[element.name]}</p></ListItem></div>

            )
        }
    }

    items = () => {
        return this.state.fields.map((element: any, index: number) => {
            return (
                <Grid key={index} style={{display: 'flex'}} className={Style.gridItems} item xs={12} sm={6}>
                    {this.inputs(element, index)}
                </Grid>

            )
        })
    }


    render() {
        return (
            <Grid className={`${Style.privateInfo} animate__animated animate__fadeIn`} container>
                {(this.props.loading) ?
                    <Loading/>
                    :
                    this.items()
                }
            </Grid>
        );
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(PrivateInfo)