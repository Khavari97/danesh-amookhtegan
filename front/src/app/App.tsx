import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
