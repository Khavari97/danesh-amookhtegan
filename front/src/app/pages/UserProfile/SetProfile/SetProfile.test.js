import React from 'react';
import Enzyme, { shallow, configure, mount } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { Provider } from "react-redux";
import App from "./SetProfile";
import renderer from "react-test-renderer";
import { render, screen, act } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Button } from '@material-ui/core';
import PrivateInfo from "./PrivateInfo/PrivateInfo";
import AboutMe from "./AboutMe/AboutMe";
import EditPrivateInfo from "./PrivateInfo/Edit/EditPrivateInfo";
import EditAboutMe from "./AboutMe/Edit/EditAboutMe";
import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store';
configure({ adapter: new Adapter() })

const middlewares = [thunk]
const mockStore = configureStore(middlewares);



describe('test set profile states', () => {
    let wrapper
    let store_mock
    const data = {
        city: "تهران",
        province: 8,
        email: "atena.ganji.ag@gmail.com",
        firstName: "آتنا",
        gender: "FEMALE",
        genderDisplay: "زن",
        lastName: "گنجی",
        phoneNumber: "+989011625458",
        university: "خوارزمی",
        username: "atena",
    }

    store_mock = mockStore({ userProfile: { loading: false, data: data } })
    store_mock.dispatch = jest.fn()
    wrapper = mount(
        <BrowserRouter>
            <Provider store={store_mock}>
                <App />
            </Provider>
        </BrowserRouter>
    );


    test('check default states', () => {
        wrapper.setState({ infoEditButton: false, aboutEditButton: false })
        expect(wrapper.find(AboutMe).exists()).toBe(true)
        expect(wrapper.find(PrivateInfo).exists()).toBe(true)
        expect(wrapper.state('infoEditButton')).toEqual(false)
        expect(wrapper.state('aboutEditButton')).toEqual(false)
    })
    test('check change states', () => {
        wrapper.setState({ infoEditButton: true, aboutEditButton: true })
        expect(wrapper.state('infoEditButton')).toEqual(true)
        expect(wrapper.state('aboutEditButton')).toEqual(true)
    })

    test('check click edit private info ', () => {
        wrapper.find('#infoEditButton').find(Button).simulate('click');

        wrapper.update()
        expect(wrapper.find(AboutMe).exists()).toBe(true)
        expect(wrapper.find(EditPrivateInfo).exists()).toBe(true)
        expect(wrapper.find(PrivateInfo).exists()).toBe(false)
    })

    test('check click edit about ', () => {
        wrapper.find('#aboutEditButton').find(Button).simulate('click');
        wrapper.update()
        expect(wrapper.find(EditAboutMe).exists()).toBe(true)
        expect(wrapper.find(EditPrivateInfo).exists()).toBe(true)

    })
    test('check click to hide edit about ', () => {
        wrapper.find('#aboutEditButton').find(Button).simulate('click');
        wrapper.update()
        expect(wrapper.find(EditAboutMe).exists()).toBe(false)
        expect(wrapper.find(EditPrivateInfo).exists()).toBe(true)

    })
    test('check click to hide edit private info ', () => {
        wrapper.find('#infoEditButton').find(Button).simulate('click');
        wrapper.update()
        expect(wrapper.find(EditAboutMe).exists()).toBe(false)
        expect(wrapper.find(EditPrivateInfo).exists()).toBe(false)
        expect(wrapper.find(AboutMe).exists()).toBe(true)
        expect(wrapper.find(PrivateInfo).exists()).toBe(true)
    })
    test('snapshot', () => {
        const snapshot = renderer.create(wrapper).toJSON();
        expect(snapshot).toMatchSnapshot();
      })

});