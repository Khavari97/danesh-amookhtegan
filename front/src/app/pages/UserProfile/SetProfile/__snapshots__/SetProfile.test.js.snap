// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test set profile states snapshot 1`] = `
<div
  className="SetProfile"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        <form
          encType="multipart/form-data"
        >
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              style={
                Object {
                  "height": "150px",
                  "width": "150px",
                }
              }
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiAvatar-fallback"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <div
              className="deleteButton"
              onClick={[Function]}
            >
              <div>
                <img
                  alt="avatar"
                  className="img"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src="delete-avatar.png"
                />
              </div>
            </div>
            <div
              className="uploadButton"
              onClick={[Function]}
            >
              <div>
                <img
                  alt="avatar"
                  className="img"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src="upload.svg"
                />
              </div>
            </div>
            <input
              accept="image/x-png,image/jpeg,image/jpg"
              className="fileUpload"
              name="newImage"
              onChange={[Function]}
              type="file"
            />
          </div>
        </form>
      </div>
    </div>
    <div
      className="MuiGrid-root header MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <img
            alt=""
            src="customer.svg"
            style={
              Object {
                "width": "30px",
              }
            }
          />
           
          <p>
            اطلاعات شخصی
          </p>
        </li>
      </div>
      <div
        className="MuiGrid-root displayFlex MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            disabled={false}
            id="infoEditButton"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <span
                  aria-hidden={true}
                  className="material-icons MuiIcon-root"
                >
                  <img
                    alt=""
                    src="pencil_2.svg"
                    style={
                      Object {
                        "width": "20px",
                      }
                    }
                  />
                </span>
              </span>
              ویرایش
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root privateInfo animate__animated animate__fadeIn MuiGrid-container"
      >
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                نام
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-testid="firstName"
              >
                آتنا
              </p>
            </li>
          </div>
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                نام خانوادگی
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-testid="lastName"
              >
                گنجی
              </p>
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                جنسیت
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-testid="genderDisplay"
              >
                زن
              </p>
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                نام کاربری
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-testid="username"
              >
                atena
              </p>
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                ایمیل
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-testid="email"
              >
                atena.ganji.ag@gmail.com
              </p>
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                استان
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-testid="province"
              >
                8
              </p>
            </li>
          </div>
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                شهر
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-testid="city"
              >
                تهران
              </p>
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                دانشگاه
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-testid="university"
              >
                خوارزمی
              </p>
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                شماره تلفن
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-testid="phoneNumber"
              >
                +989011625458
              </p>
            </li>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root header MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <img
            alt=""
            src="new_topic.svg"
            style={
              Object {
                "width": "30px",
              }
            }
          />
            
          <p>
            درباره ی من
          </p>
        </li>
      </div>
      <div
        className="MuiGrid-root displayFlex MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            disabled={false}
            id="aboutEditButton"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <span
                  aria-hidden={true}
                  className="material-icons MuiIcon-root"
                >
                  <img
                    alt=""
                    src="pencil_2.svg"
                    style={
                      Object {
                        "width": "20px",
                      }
                    }
                  />
                </span>
              </span>
              ویرایش
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root animate__animated animate__fadeIn MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
        >
          <div
            className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
          >
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
                data-testid="desc"
              />
            </li>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
